repair-missing/script.whizzml

```diff
@@ -0,0 +1,192 @@
+;; This is a hack:  Field IDs are not conserved at batch prediction time, so
+;; we need to make sure that all of the fields have assigned IDs exactly as
+;; batch prediction will assign them (which essentially amounts to putting
+;; generated fields at the end).  To do this, we make a dummy model, do a dummy
+;; batch prediction, then remove the dummy column.
+(define (flatten-dataset dataset-id)
+  (let (fds ((fetch dataset-id) "fields"))
+    (if (some (lambda (fid) (= ((fds fid) "optype") "datetime")) (keys fds))
```

Comment:
There's almost certainly something *I'm* missing.  I thought datetimes were the only time we were generating fields, but you're saying there are others?  What are they?